app/models/order_detail.rb

```diff
@@ -716,16 +722,16 @@ def in_dispute?
   end
 
   def disputed?
+    # only used in specs
     dispute_at.present? && !canceled?
   end
 
-  def cancel_reservation(canceled_by, order_status: OrderStatus.canceled_status, admin: false, admin_with_cancel_fee: false)
-    res = reservation
-    res.canceled_by = canceled_by.id
+  def cancel_reservation(canceled_by, canceled_reason: nil, order_status: OrderStatus.canceled_status, admin: false, admin_with_cancel_fee: false)
+    self.canceled_by = canceled_by.id
+    self.canceled_reason = canceled_reason
 
     if admin
-      res.canceled_at = Time.zone.now
-      return false unless res.save
+      return false unless update_attributes(canceled_at: Time.current)
```

Comment:
I'm not sure why this method is structured the way it is--I mostly didn't change it, just updated it to work with the new schema.  I thought about refactoring it more extensively, but I'm a little afraid of breaking it--I personally can't see why the save should fail, but I also don't see where the later save is that you're referring to...